pkg/trace/agent/run.go

```diff
@@ -145,13 +145,13 @@ func Run(ctx context.Context) {
 
 	rand.Seed(time.Now().UTC().UnixNano())
 
-	// Start workload metadata store before tagger
-	workloadmeta.GetGlobalStore().Start(context.Background())
-
 	var t tagger.Tagger
 	if coreconfig.Datadog.GetBool("apm_config.remote_tagger") {
 		t = remote.NewTagger()
 	} else {
+		// Start workload metadata store before tagger
```

Comment:
Just a nitty comment: is this required to be started before `local.NewTagger()`? If yes, shouldn't we change the API of `local.NewTagger()` instead to ensure that? I feel it might make it easier to introduce a programming error easily here if anyone were to mix up the order of things.

Reply:
the `workloadmeta.GetGlobalStore()` is/will be used also by the autodiscovery component. That is why we have 2 separate functions to init them.

We can considere when the migration is done to the workloadmetadata store, to update the `NewTagger` function signature to provide the instance pointer of the metadatastore, to make it more obvious. I will write a card in our backload.